spec/features/staff/users_can_create_a_refund_spec.rb

```diff
@@ -28,6 +28,12 @@
         expect(page).to have_content("£100")
       end
     end
+
+    scenario "must supply the required information to create a refund" do
+      given_i_am_on_the_new_refund_form
```

Comment:
I am interested in your view on this style of writing specs, I can see how valuable it is, but I wonder if there is something to be said for consistency across the test suite as well? 

To be clear, I don't think we need to change this, I am interested in you view!

Reply:
I think we should be aiming to write our feature tests in a way which describes the behaviour of the code from the point of view of the user.

 e.g. I'm being shown error messages so that I can post the form successfully:

`then_i_expect_to_see_how_i_need_to_correct_the_refund_form`

When reviewing this spec, the purpose (the business benefit) of the code we're describing is clear. Clearer than if we make the constituent assertions:

```
expect(page).to have_content("Select a financial quarter")
expect(page).to have_content("Select a financial year")
expect(page).to have_content("Enter a refund amount")

```

without the commentary provided by the enclosing method name `then_i_expect_to_see_how_i_need_to_correct_the_refund_form`.

I see this style (extracting Given/When/Then steps into clearly named methods) as a way to incrementally improve the clarity and maintainability of BEIS' large estate of RSpec/Capybara feature specs. Of course it's a balance between change and consistency, but here I'm making the case for a using a more behaviour-centric style **within the existing convention** of RSpec/Capybara features.